src/vm.rs

```diff
@@ -683,18 +684,52 @@ impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
     }
 
     // Apply rewards (balance increments) to beneficiary accounts, etc.
-    fn apply_rewards(&self, write_set: &mut WriteSet, tx: &TxEnv, gas_used: U256) {
+    fn apply_rewards<DB: Database>(
+        &self,
+        write_set: &mut WriteSet,
+        tx: &TxEnv,
+        gas_used: U256,
+        #[allow(unused_variables)] evm_context: &EvmContext<DB>,
+    ) {
+        let mut gas_price = if let Some(priority_fee) = tx.gas_priority_fee {
+            std::cmp::min(tx.gas_price, priority_fee + self.block_env.basefee)
+        } else {
+            tx.gas_price
+        };
+        if self.spec_id.is_enabled_in(SpecId::LONDON) {
+            gas_price = gas_price.saturating_sub(self.block_env.basefee);
+        }
+
         let rewards: SmallVec<[(MemoryLocationHash, U256); 1]> = match self.reward_policy {
             RewardPolicy::Ethereum => {
-                let mut gas_price = if let Some(priority_fee) = tx.gas_priority_fee {
-                    std::cmp::min(tx.gas_price, priority_fee + self.block_env.basefee)
+                smallvec![(self.beneficiary_location_hash, gas_price * gas_used)]
+            }
+            #[cfg(feature = "optimism")]
+            RewardPolicy::Optimism {
```

Comment:
A link to these fee formulas would help.

Reply:
added a link to @bluealloy/revm